README.md

```diff
@@ -1,37 +1,91 @@
+- [clean_tables.cwl](cwl/clean_tables.cwl): For all input tables (picard metrics, gene counts, star log), this tool converts Synapse IDs to specimen IDs and removes samples that are duplicates (specimen ID and values must be identical to be removed). The script can be found [here](bin/clean_tables.py).
 
 ## Dockerfile
 
-A Dockerfile should exist in the top level directory of this repository.
+A Dockerfile exists in the top level directory of this repository. Builds are automatically triggered on pushes in which the most recent commit message does not contain the `[skip-ci]` string.
 
 ## CWL
 
-A directory [cwl](cwl) should contain tool definitions in the CWL language.
+The [cwl](cwl) folder contains tool definitions in the Common Workflow language.
 
 ## Tests
 
-Tests for the CWL tool should be added to the [tests](tests) directory.
+[`cwltest`](https://github.com/common-workflow-language/cwltest) is used for
+testing tools. Tests can be added in `tests/test-descriptions.yaml`. Tests are automatically performed on pushes in which the most recent commit message does not contain the `[skip-ci]` string.  When tests pass, a docker image is published to dockerhub, and the cwl files are updated to point to this new build.
+
+# Contributing
+
+To contribute changes to the dockerfile or CWL tools, please create a fork of this repository and develop on a feature branch. Pull requests will be reviewed before merging into this repository.
 
 ## Continuous Deployment and Versioning
 
-This template uses GitHub actions to perform automated versioning, version bumping, building of tagged Docker images, and pushing images to DockerHub.
+This template uses GitHub actions to perform automated versioning, version
+bumping, building of tagged Docker images, and pushing images to DockerHub.
+
+### CI
+Defined in [.github/workflows/ci.yaml](.github/workflows/ci.yaml), this action
+runs on each push to master where the commit does not contain '[skip-ci]'.
+
+### TAG CI
+Defined in [.github/workflows/ci.yaml](.github/workflows/tag-ci.yaml), this action
+runs on each tag push.
 
 ### Credentials
 
-This uses GitHub secrets to store credentials for the GitHub action to push to the `sagebionetworks` DockerHub account using a service account. It also uses a service GitHub account for commiting new version automatically. All repositories that are generated from this template will need to have this service account added to it.
+This uses GitHub secrets to store credentials for the GitHub action to push to
+the `sagebionetworks` DockerHub account using a service account. All repositories
+that are generated from this template will need to have this service account
+added to it.
 
 ### DockerHub
 
-Ideally, the name of the DockerHub repository should be the same as the GitHub repository. The workflow uses this name by default as the name of the DockerHub branch to push to.
+Ideally, the name of the DockerHub repository should be the same as the GitHub
```

Comment:
Change to something like "The workflow pushes to a Dockerhub repository with the same name as this github repository". There are no branches in Dockerhub.